BeeSwift/JSONGoal.swift

```diff
@@ -614,11 +614,12 @@ class JSONGoal {
     func updateBeeminderWithValue(datapointValue : Double, daystamp : String, success: (() -> ())?, errorCompletion: (() -> ())?) {
         
```

Comment:
As the code is written, we should probably be checking here too that this goal is still associated with a healthkit metric (and even the same one for which the query was created). This might be relevant to the issue reported where a newly created goal of the same name as one previously seen in the app was provided with HK data, apparently from the previous association (in the deleted goal of the same slug) - #23